jumanji/environments/packing/job_shop/improvement/scheduling/registry.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from typing import List, Type
+
+from jumanji.environments.packing.job_shop.improvement.scheduling.base import (
+    AbstractSchedulingMethod,
+)
+
+# Store method classes in order of registration
+_REGISTRY = OrderedDict()
+
+
+class MethodRegistry:
+    """Registry for scheduling methods with automatic discovery and ordering."""
+
+    @staticmethod
+    def register(cls: Type[AbstractSchedulingMethod]) -> Type[AbstractSchedulingMethod]:
+        """Register a scheduling method class.
+
+        Args:
+            cls: The scheduling method class to register.
+
+        Returns:
+            The method class (for use as a decorator).
+
+        Raises:
+            ValueError: If method name is already registered.
+        """
+        # Create temporary instance to get name
+        temp_instance = cls(num_jobs=1, num_machines=1, max_num_jobs=1, max_num_ops=1)
```

Comment:
Could we make the name a class attribute to avoid having to instantiate the class to read the name ?